provider/pkg/provider/provider.go

```diff
@@ -524,3 +532,298 @@ func convertInterfaceToPropertyValue(v interface{}) resource.PropertyValue {
+		}
+	}
+
+	// Build output (simplified team info)
+	teamProps := make([]resource.PropertyValue, len(userTeams))
+	for i, team := range userTeams {
+		teamProps[i] = resource.NewObjectProperty(resource.PropertyMap{
+			"kind":        resource.NewStringProperty(team.Type),
+			"name":        resource.NewStringProperty(team.Name),
+			"displayName": resource.NewStringProperty(team.DisplayName),
+			"description": resource.NewStringProperty(team.Description),
+		})
+	}
+
+	outputProps := resource.PropertyMap{
+		"teams": resource.NewArrayProperty(teamProps),
+	}
+
+	outputProperties, err := plugin.MarshalProperties(outputProps, plugin.MarshalOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return &pulumirpc.InvokeResponse{
+		Return: outputProperties,
+	}, nil
+}
+
+// invokeFunctionGetStacks implements the getStacks function
+func (k *pulumiserviceProvider) invokeFunctionGetStacks(ctx context.Context, req *pulumirpc.InvokeRequest) (*pulumirpc.InvokeResponse, error) {
+	if k.client == nil {
+		return nil, fmt.Errorf("provider not configured")
+	}
+
+	// Parse inputs
+	inputs, err := plugin.UnmarshalProperties(req.GetArgs(), plugin.MarshalOptions{KeepUnknowns: true, SkipNulls: true})
+	if err != nil {
+		return nil, err
+	}
+
+	var maxResults *int
+	if inputs["maxResults"].HasValue() && inputs["maxResults"].IsNumber() {
+		mr := int(inputs["maxResults"].NumberValue())
+		maxResults = &mr
+	}
+
+	// Call the API
+	response, err := k.client.ListUserStacks(ctx, maxResults)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list user stacks: %w", err)
+	}
+
+	// Build output
+	stackProps := make([]resource.PropertyValue, len(response.Stacks))
+	for i, stack := range response.Stacks {
+		props := resource.PropertyMap{
+			"id":          resource.NewStringProperty(stack.ID),
+			"orgName":     resource.NewStringProperty(stack.OrgName),
+			"projectName": resource.NewStringProperty(stack.ProjectName),
+			"stackName":   resource.NewStringProperty(stack.StackName),
+		}
+		if stack.LastUpdate != nil {
+			props["lastUpdate"] = resource.NewNumberProperty(float64(*stack.LastUpdate))
+		}
+		if stack.ResourceCount != nil {
+			props["resourceCount"] = resource.NewNumberProperty(float64(*stack.ResourceCount))
+		}
+		stackProps[i] = resource.NewObjectProperty(props)
+	}
+
+	outputProps := resource.PropertyMap{
+		"stacks": resource.NewArrayProperty(stackProps),
+	}
+
+	if response.ContinuationToken != nil {
+		outputProps["continuationToken"] = resource.NewStringProperty(*response.ContinuationToken)
+	}
+
+	outputProperties, err := plugin.MarshalProperties(outputProps, plugin.MarshalOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return &pulumirpc.InvokeResponse{
+		Return: outputProperties,
+	}, nil
+}
+
+// invokeFunctionGetStackPermissions implements the getStackPermissions function
+func (k *pulumiserviceProvider) invokeFunctionGetStackPermissions(ctx context.Context, req *pulumirpc.InvokeRequest) (*pulumirpc.InvokeResponse, error) {
```

Comment:
Why does this one need to go into one? It performs two API calls that do something slightly different, maybe it could map more directly to how our API looks?

Also, I'm not sure how this works, but does fetching collaborators return users that have access to the stack on their own, or do they have access by virtue of being part of a team? In the latter case, is it relevant to know by which team membership they have access?